src/main/java/com/github/matidominati/medicalclinic/repository/PatientRepositoryImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.github.matidominati.medicalclinic.repository;
+
+import com.github.matidominati.medicalclinic.personInfo.Patient;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class PatientRepositoryImpl implements PatientRepository {
+    private final List<Patient> patients = new ArrayList<>();
+
+    @Override
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    @Override
+    public Patient getPatient(String email) {
+        Optional<Patient> patient = findPatientByEmail(email);
+        if (patient.isEmpty()) {
+            throw new RuntimeException("Patient with the provided email does not exist");
```

Comment:
zwroc pustego optionala, nie rzucaj wyjatku. Zostaw odpowiedzialnosc za powiedzenie co ma sie zadziac w przypadku pustego optionala komus kto to wywoluje